app/javascript/components/KeywordSearch.js

```diff
@@ -1,39 +1,41 @@
-import React from 'react'
+import React, { useState, useContext } from 'react';
 import Button from 'react-bootstrap/lib/Button'
 import InputGroup from 'react-bootstrap/lib/InputGroup'
 import Form from 'react-bootstrap/lib/Form'
 import { faSearch } from '@fortawesome/free-solid-svg-icons'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { StudySearchContext } from 'components/search/StudySearchProvider'
 
 /**
  * Component to search using a keyword value
- *
+ * optionally takes a 'keywordValue' prop with the initial value for the field
  */
 export default function KeywordSearch(props) {
-  const handleSubmit = event => {
+  const searchContext = useContext(StudySearchContext)
+  const [keywordValue, setKeywordValue] = useState('');
+
+  const handleSubmit = submitValue => {
```

Comment:
The above style is more common in class components, but in function components it would conventionally be:

```suggestion
  function handleSubmit(submitValue) {
```

At least as I see in React docs, e.g. https://reactjs.org/docs/handling-events.html.  Its traditional form is also slightly more readable, at least for me.